packages/plugin-electron-deliver-minidumps/send-minidump.js

```diff
@@ -0,0 +1,77 @@
+const { createReadStream } = require('fs')
+const { createGzip } = require('zlib')
+const { basename } = require('path')
+const payload = require('@bugsnag/core/lib/json-payload')
+const FormData = require('form-data')
+
+module.exports = (net, client) => {
+  const send = (opts, formData) => {
+    return new Promise((resolve, reject) => {
+      const req = net.request(opts, response => {
+        if (isOk(response)) {
+          resolve()
+        } else {
+          const err = new Error(`Bad status code from API: ${response.statusCode}`)
+          err.isRetryable = isRetryable(response.statusCode)
+          reject(err)
+        }
+      })
+
+      req.on('error', reject)
+
+      try {
+        formData.pipe(req)
+      } catch (err) {
+        // if we can't write this body to the request, it's likely impossible to
+        // ever send it successfully
+        err.isRetryable = false
+        reject(err)
+      }
+    })
+  }
+
+  const sendMinidump = async (minidumpPath, event) => {
+    const url = new URL(client._config.endpoints.minidumps)
+    url.searchParams.set('api_key', client._config.apiKey)
```

Comment:
I don't see anything mentioning this in the design but an event can have a different API key to the config's API key. This lets you send different events to different Bugsnag projects, e.g. if parts of the code are owned by different teams

Should we be using the event API key instead of the config API key if there is an event?

Reply:
Raised this in Slack so don't worry about changing anything yet

Reply:
> I believe that the API key in the query parameter would take precedence over the one in the payload so they could probably be different (if it is a pain to get them the same) but ideally both would match

```suggestion
    url.searchParams.set('api_key', event.apiKey || this._config.apiKey)
```

Would be good to add a test for this case as well

Reply:
Will be done separately